### PR TITLE
HADOOP-18073. Upgrade S3A in 3.3 branch to AWS SDK V2.(#5707)

### PR DESCRIPTION
**Pull request by steveloughran**
### Description of PR

rebase of feature-HADOOP-18073-s3a-sdk-upgrade-3.3-rebase onto branch-3.3

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by mukund-thakur** [comment]
Shouldn't there be one for trunk first?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  12m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  3s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 79 new or modified test files.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 38s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  40m 40s |  |  branch-3.3 passed  |
| +1 :green_heart: |  compile  |  20m 12s |  |  branch-3.3 passed  |
| +1 :green_heart: |  checkstyle  |   3m 10s |  |  branch-3.3 passed  |
| +1 :green_heart: |  mvnsite  |   3m 14s |  |  branch-3.3 passed  |
| +1 :green_heart: |  javadoc  |   2m  3s |  |  branch-3.3 passed  |
| +0 :ok: |  spotbugs  |   0m 41s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  41m 38s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 58s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  20m 12s |  |  the patch passed  |
| -1 :x: |  javac  |  20m 12s | [/results-compile-javac-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/artifact/out/results-compile-javac-root.txt) |  root generated 17 new + 1862 unchanged - 15 fixed = 1879 total (was 1877)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   3m  5s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 17 new + 73 unchanged - 7 fixed = 90 total (was 80)  |
| +1 :green_heart: |  mvnsite  |   3m 10s |  |  the patch passed  |
| -1 :x: |  javadoc  |   0m 41s | [/results-javadoc-javadoc-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/artifact/out/results-javadoc-javadoc-hadoop-tools_hadoop-aws.txt) |  hadoop-tools_hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +0 :ok: |  spotbugs  |   0m 31s |  |  hadoop-project has no data from spotbugs  |
| -1 :x: |  spotbugs  |   1m 35s | [/new-spotbugs-hadoop-tools_hadoop-aws.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/artifact/out/new-spotbugs-hadoop-tools_hadoop-aws.html) |  hadoop-tools/hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  shadedclient  |  41m 53s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 29s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |  18m 52s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 54s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 58s |  |  The patch does not generate ASF License warnings.  |
|  |   | 252m 18s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-aws |
|  |  Inconsistent synchronization of org.apache.hadoop.fs.s3a.S3AFileSystem.s3AsyncClient; locked 60% of time  Unsynchronized access at S3AFileSystem.java:60% of time  Unsynchronized access at S3AFileSystem.java:[line 1764] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5858 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets xmllint markdownlint |
| uname | Linux 90f724bc2e2e 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / aba41830212c39f5a96928b721997cb3717cc0d0 |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~18.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/testReport/ |
| Max. process+thread count | 3144 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/1/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
@mukund-thakur this is the rebase for the feature branch-3.3; I do have one for trunk too, #5857

i want to make sure everyone is happy with this not complicating any open PRs, and I want to rebase atop branches with #5821 in as well.

**Comment by steveloughran** [comment]
will manually merge if yetus is happy

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  4s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 79 new or modified test files.  |
|||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 26s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  40m 18s |  |  branch-3.3 passed  |
| +1 :green_heart: |  compile  |  19m 22s |  |  branch-3.3 passed  |
| +1 :green_heart: |  checkstyle  |   2m 59s |  |  branch-3.3 passed  |
| +1 :green_heart: |  mvnsite  |   3m 44s |  |  branch-3.3 passed  |
| +1 :green_heart: |  javadoc  |   2m 31s |  |  branch-3.3 passed  |
| +0 :ok: |  spotbugs  |   0m 52s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  38m 11s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   1m  3s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  2s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 35s |  |  the patch passed  |
| -1 :x: |  javac  |  18m 35s | [/results-compile-javac-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/artifact/out/results-compile-javac-root.txt) |  root generated 17 new + 1862 unchanged - 15 fixed = 1879 total (was 1877)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   2m 53s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 17 new + 73 unchanged - 7 fixed = 90 total (was 80)  |
| +1 :green_heart: |  mvnsite  |   3m 43s |  |  the patch passed  |
| -1 :x: |  javadoc  |   0m 55s | [/results-javadoc-javadoc-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/artifact/out/results-javadoc-javadoc-hadoop-tools_hadoop-aws.txt) |  hadoop-tools_hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +0 :ok: |  spotbugs  |   0m 44s |  |  hadoop-project has no data from spotbugs  |
| -1 :x: |  spotbugs  |   1m 43s | [/new-spotbugs-hadoop-tools_hadoop-aws.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/artifact/out/new-spotbugs-hadoop-tools_hadoop-aws.html) |  hadoop-tools/hadoop-aws generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  shadedclient  |  39m 29s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 43s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |  18m 22s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  8s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  9s |  |  The patch does not generate ASF License warnings.  |
|  |   | 234m 28s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-tools/hadoop-aws |
|  |  Inconsistent synchronization of org.apache.hadoop.fs.s3a.S3AFileSystem.s3AsyncClient; locked 60% of time  Unsynchronized access at S3AFileSystem.java:60% of time  Unsynchronized access at S3AFileSystem.java:[line 1764] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5858 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets xmllint markdownlint |
| uname | Linux 2babfb97867e 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / d4deb9bebaeb7396deb8f1b78757f1c0540a6515 |
| Default Java | Private Build-1.8.0_362-8u372-ga~us1-0ubuntu1~18.04-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/testReport/ |
| Max. process+thread count | 1302 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5858/2/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java [hunk 1]
```diff
@@ -1215,213 +1079,6 @@ public static void deleteWithWarning(FileSystem fs,
...
-   * Initializes AWS SDK proxy support in the AWS client configuration
-   * if the S3A settings enable it.
-   *
-   * @param conf Hadoop configuration
-   * @param bucket Optional bucket to use to look up per-bucket proxy secrets
-   * @param awsConf AWS SDK configuration to update
-   * @throws IllegalArgumentException if misconfigured
-   * @throws IOException problem getting username/secret from password source.
-   */
-  public static void initProxySupport(Configuration conf,
```
Comment (mukund-thakur) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java:1351]: @ahmarsuhail  why this has been removed? 
CC @steveloughran

Comment (ahmarsuhail) [reply hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java:1351]: All config stuff (including this) has been moved to imp/AWSClientConfig. Proxy config can be found is AWSClientConfig.createProxyConfiguration()

Comment (mukund-thakur) [reply hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java:1351]: Thanks

**Comment by steveloughran** [comment]
ok, done the forced push of this; closing

**Pull request closed without merging**